build.gradle

```diff
@@ -969,3 +972,84 @@ task releaseAndTest {
+ * @param jar fully qualified file name of a jar file
+ * @param pom pom file describing the artifact
+ */
+void mvnDeployToLocalRepo(String jar, String pom) {
+    exec {
+        executable 'mvn'
+        args += [
+                'install:install-file',
+                "-Dfile=${jar}",
+                "-DpomFile=${pom}",
+                // TODO: This changes the version number of the artifact, but not the version number in the pom file.
+//                "-Dversion=${version}",
+                '-DgeneratePom=true'
+
+        ]
+    }
+}
+
+/**
+ * Deploys the jar file produced by the "jar" task of the given project to the local Maven repository.
+ * @param project project whose jar file is deployed
+ * @param pom pom file describing the artifact
+ */
+void mvnDeployToLocalRepo(Project project, String pomFile) {
+    String jarFile = project.tasks.getByName('jar').archiveFile.get().toString();
+    mvnDeployToLocalRepo(jarFile, pomFile)
+}
+
+/**
+ * Signs and deploys the given jar file to the Sonatypes Maven repository.
+ * @param jar fully qualified file name of a jar file
+ * @param pom pom file describing the artifact
+ * @param classifier the kind of artifact; sources or javadoc; defaults to the empty string
+ */
+void mvnSignAndDeployToSonatype(String jar, String pom, String classifier="") {
+    String passwordFile = "/projects/swlab1/checker-framework/hosting-info/release-private.password"
+    String passphrase = new File(passwordFile).readLines().get(0);
+
+    exec {
+        executable 'mvn'
+        args += [
+                'gpg:sign-and-deploy-file',
+                '-Durl=https://oss.sonatype.org/service/local/staging/deploy/maven2/',
+                '-DrepositoryId=sonatype-nexus-staging',
+                "-DpomFile=${pom}",
+                "-Dfile=${jar}",
+                '-Dgpg.keyname=checker-framework-dev@googlegroups.com',
+                "-Dgpg.passphrase=${passphrase}"
+        ]
+        if (!classifier.isEmpty()) {
+            args += ["-Dclassifier=${classifier}"]
+        }
+    }
+}
+
+/**
+ * Signs and deploys multiple jar files for a single artifact.
+ * @param classesJar class jar file
+ * @param sourceJar source jar file corresponding to the classesJar
+ * @param javadocJar javadoc jar file corresponding to the classesJar
+ * @param pom pom file describing the artifact
+ */
+void mvnSignAndDeployMultipleToSonatype(String classesJar, String sourceJar, String javadocJar, String pom) {
+    mvnSignAndDeployToSonatype(classesJar, pom)
+    mvnSignAndDeployToSonatype(sourceJar, pom, "sources")
+    mvnSignAndDeployToSonatype(javadocJar, pom,"javadoc")
+}
+
+/**
+ * Signs and deploys the classes jar, sources jar, and javadoc jar files for a single artifact.
+ * @param project with a jar, sourcesJar, and javadocJar tasks defined
+ * @param pom pom file describing the artifact
+ */
+void mvnSignAndDeployMultipleToSonatype(Project project, String pom) {
```

Comment:
This could call the method that is defined on line 1040.  That wouldn't make the code much shorter, but it would clarify the relationship between the two methods, which would helped me to understand them.  The documentation could also be made more similar -- the difference in the Javadoc first sentence also made me think the methods were more different than they are.